Document how to specify a static resource bundle to load as the last … (#1802)
plone/documentation: docs/classic-ui/static-resources.md

  </registry>
```

> You can also register a JavaScript file and a CSS file in the same bundle.

Reviewer: "It is also possible to include both a JavaScript file and a CSS file in the same bundle." If you don't want to use "you".
```suggestion
It is also possible to include both a JavaScript file and a CSS file in the same bundle.
```

Author: I want to address the reader with the active voice of "you can", not the passive voice of "it is possible". The former tone is more encouraging and friendly to the reader. It's a thing we adopted from the [Microsoft Writing Style Guide](https://learn.microsoft.com/en-us/style-guide/grammar/verbs#active-and-passive-voice), as noted in https://6.docs.plone.org/contributing/documentation/authors.html#plone-documentation-styleguide.

## Available attributes

The following attributes are available for registering a static resource.

`enabled`
:   Boolean.
    Whether the bundle is enabled or not.
    If it is disabled, the bundle will not be loaded.

`jscompilation`
:   String.
    The path to the compiled JavaScript file.

`csscompilation`
:   String.
    The path to the compiled CSS file.

`depends`
:   String.
    A comma-separated list of bundles that this bundle depends on.
    For a single dependency, do not insert commas.

    When a bundle depends on another one, its `<script>` or `<link>` tag is rendered after the bundle it depends on.

    If the bundle's dependencies do not exist, then the bundle will not render.

    The `depends` attribute may be assigned the value of `all`, making this bundle render last, after all other bundles.
    The `all` value lets you load CSS files after the automatically added theme resources and override CSS declarations from your own custom CSS files.

>     If you set multiple bundles to `all`, then these bundles will render in the order in which you define them.

Reviewer 2: ```suggestion
    If you set multiple bundles to `all`, then these bundles will render in alphabetical order by its name.
```

## Loading order of resources

You can use the `depends` attribute to define the order in which resources load.

You can specify a list of either the names of the bundles on which this bundle depends or a single list item of `["all"]`.

> In the latter case of `["all"]` for a bundle, all other JavaScript and CSS resources in their resource groups will render before this bundle, making it the final bundle to render in its resource group.

Reviewer 3: What happens if there are two bundles depending on "all"? Is this prevented? If not, is there some order or is it random?

Reviewer 2: I'd say then the alphabetical sorting in the resource registry takes effect, which you could manipulate with special prefixes (eg. "zzz_mybundle")

Reviewer 4: ~~If more bundles depend on "all", all the "all" bundles are rendered in the order they were defined~~

That was apparently wrong. @petschki answer above is correct.

Author: Addressed in f80c6103029f37c3e9bfb9c5fc70292f36442881

> You can specify a list of either the names of the bundles on which this bundle depends or a single list item of `["all"]`.
> In the latter case of `["all"]` for a bundle, all other JavaScript and CSS resources in their resource groups will render before this bundle, making it the final bundle to render in its resource group.

Reviewer 4: ```suggestion
You can specify a comma separated list of either the names of the bundles on which this bundle depends or the special value `"all"`.
In the latter case of `"all"` for a bundle, all other JavaScript and CSS resources in their resource groups will render before this bundle, making it the final bundle to render in its resource group.
If multiple bundles use the `"all"` dependency, they are all output at the end, in the order they were defined.
```

Author: This content was redundant after merging your content, so I removed it.

This includes the theme, such as Barceloneta's theme CSS.

This feature lets you override a theme with custom CSS from a bundle.

> Previously you had to add the CSS customizations to the registry via the `custom_css` settings, then update the registry after every change.

Reviewer 4: ```suggestion
Note: You can also add CSS customizations through-the-web via the theming control panel.
There you can add some custom CSS.
These custom CSS file is always rendered last, also after any bundles with a dependency of "all"
```

Author: I merged content and rewrote this one.

```{versionadded} Plone 6.0.14

> The `depends` attribute may be assigned the value of `["all"]`, making this bundle render last, after all other bundles.

Reviewer 4: ```suggestion
The `depends` attribute may be assigned the value of `"all"`, making this bundle render last, after all other bundles.
```

```

```{note}

> Setting `depends=["all"]` does not affect custom CSS that you define in the {guilabel}`Theming` control panel, which _always_ renders as the last style resource.

Reviewer 4: ```suggestion
Setting `depends="all"` does not affect custom CSS that you define in the {guilabel}`Theming` control panel, which _always_ renders as the last style resource.
```

Author: I realized that we use XML syntax in the examples, so I changed these to be plain old inline literals.
